crates/model/Cargo.toml

```diff
@@ -26,6 +26,7 @@ serde_json = { workspace = true }
 serde_with = { workspace = true }
 strum = { workspace = true }
 web3 = { workspace = true, features = ["signing"] }
+database = { workspace = true }
```

Comment:
The `model` crate now depending on `database` doesn't seem great. 🤔 
Conceptually they should be fairly independent. `model` is mostly for passing around data whereas `database` is for implementing the DB SQL queries. Passing around data should not rely on logic that implements DB logic.

If anything it seems like the dependency should be the other way around. But overall I think it might be the most reasonable to have conversion functions in shared. Since both types of the conversion are defined outside of `shared` we can't use conversion traits unfortunately.

Reply:
I feel we should move towards conversion traits since most of the code is conversions.
But that seems to be a lot of changes if I have to invert the dependency

Reply:
Hi, I raised a [different MR](https://github.com/cowprotocol/services/pull/3325) not touching the project structure. 
Please take a look and let me know your thoughts.